ext/featurefmt/rpm/contentmanifest/updater.go

```diff
@@ -0,0 +1,170 @@
+package contentmanifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/quay/clair/v3/pkg/httputil"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	lastUpdateDate  = time.Date(1970, time.January, 1, 1, 1, 1, 1, time.UTC)
+	lastHeaderQuery = time.Date(1970, time.January, 1, 1, 1, 1, 1, time.UTC)
+
+	mappingFileMutex = sync.Mutex{}
+	updaterMutex     = sync.Mutex{}
+)
+
+// LocalUpdaterJob periodically updates mapping file and store it in local storage
+type LocalUpdaterJob struct {
+	LocalPath string
+	URL       string
+}
+
+// Get translate repositories into CPEs using a mapping file
+func (updater *LocalUpdaterJob) Get(repositories []string) ([]string, error) {
+	mappingFileMutex.Lock()
+	defer mappingFileMutex.Unlock()
+	f, err := os.Open(updater.LocalPath)
```

Comment:
I wouldn't hold the lock open for the opening of the file, open and close the lock just where you updating the critical area.

Reply:
I lock the file mutex before `Open` and unlock it right after `Close`. What would you change?

Reply:
Oh I see, why not just open the file with READONLY flag?
https://golang.org/src/os/file.go?s=9377:9443#L295

You don't need to lock the file access if you are only reading.